app/helpers/katello/katello_urls_helper.rb

```diff
@@ -1,15 +1,27 @@
 require 'uri'
+
 module Katello
   module KatelloUrlsHelper
+    def foreman_settings_url
+      Setting[:foreman_url].sub(/^https/, 'http')
+    end
+
     def subscription_manager_configuration_url(host = nil, rpm = true)
       prefix = if host && host.content_source
-                 "http://#{@host.content_source.hostname}"
+                 "http://#{host.content_source.hostname}"
                else
-                 Setting[:foreman_url].sub(/\Ahttps/, 'http')
+                 foreman_settings_url
                end
       config = rpm ? SETTINGS[:katello][:consumer_cert_rpm] : SETTINGS[:katello][:consumer_cert_sh]
 
       "#{prefix}/pub/#{config}"
     end
+
+    def repository_url(content_path, schema = 'http')
+      path = ::Katello::Glue::Pulp::Repos.repo_path_from_host(@host, content_path)
+      return content_path if content_path.start_with?('http', 'https', 'ftp', 'ftps')
+      return "#{foreman_settings_url}/pulp/repos/#{path}" unless @host.content_source
```

Comment:
I guess the `foreman_settings_url` should also take the `schema` as argument, to be consistent with the rest of this method.

Reply:
`foreman_settings_url` is the default setting, and already has the schema, are suggestion to replace the schema of `foreman_settings_url` with the provided `schema`?

Reply:
I would expect `def foreman_setting_url(schema = 'http')` and using the schema argument rather than having that hardcoded in the `foreman_setting_url` method.